### ISSUE TITLE
When "enable gzip compression = no" in netdata.conf, I get no response.

### ISSUE THREAD

**Issue by toalex77** [firehol/netdata#1956]
I've installed netdata on my Raspberry Pi 3.
As said in the title, when I set "enable gzip compression = no" in netdata.conf, I get no response from netdata web server and the clients (Chrome or Firefox it does not do any difference) waiting to load the page forever.

**Comment by ktsaou** [comment]
hi.

Just tried it on a raspberry pi 2 and it works.
Are you sure?

**Comment by toalex77** [comment]
This is my netdata.conf (without comment lines):

```
[global]
        history = 3600
        memory deduplication (ksm) = no
        debug log = none
        error log = none
        access log = none

[web]
        bind to = 192.168.0.48
        mode = single-threaded

[plugins]
        tc = no
        cgroups = no
        enable running new plugins = no
        charts.d = no
        fping = no
        node.d = no

[plugin:proc]
        /sys/kernel/mm/ksm = no
        /proc/net/snmp6 = no
        /proc/net/ip_vs/stats = no
        /proc/net/stat/conntrack = no
        /proc/net/stat/synproxy = no
```
Can it be a combination with other options?

**Comment by ktsaou** [comment]
I just tried them. It works. I can't reproduce it.
(btw, this config has not disabled gzip compression - I tried with and without it though)

Could you please try to switch back to the multi-threaded web server and retry?
Does it also happen when multi-threaded?

**Comment by toalex77** [comment]
Yes, I know, the configuration that I've previously posted is the one without the "enable gzip compression = no", so netdata works.

It seems that the combination of "mode = single-threaded" and "enable gzip compression = no" cause the issue.

**Comment by ktsaou** [comment]
hm... I can't reproduce this though...

When this happens, could you please run `top` to check netdata? How much CPU does it use?

**Comment by toalex77** [comment]
When this happens, the CPU usage of netdata is below 2.0%, and it seems it does not collect any data (there is a hole in graphs when this issue appear).

**Comment by ktsaou** [comment]
Which graphs have the hole? I think this is a deadlock inside netdata, so I need your help to narrow it down.

**Comment by meepmeep** [comment]
Today I noticed that me netdata doesn't work anymore (see attached capture below).

I'm using netdata behind lighttpd, so gzip compression is disable in netdata ([wiki](https://github.com/firehol/netdata/wiki/Running-behind-lighttpd)). 
If I activate it again (enable gzip compression = yes), it works and netdata loads successfully. 

(I tried with _enable gzip compression = no_  in WEB section, and _enable web responses gzip compression = no_ in Global section, same result)

![2018-01-16-13-01-10 7 0 200_19999](https://user-images.githubusercontent.com/584651/34988802-4dd80f08-fac0-11e7-991a-a804a1243ee6.png)
